### PR TITLE
Sort the linked list using quicksort technique

### PR DESCRIPTION
**Pull request by Darpan-Balar**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [ ] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: data_structures/linked_list/quicksort_linkedlist.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        """
+        Pushes the element at the start of the linkedlist and returns None
+        """
+        new_node = Node(new_data)
+        new_node.next = self.head
+        self.head = new_node
+        return
+
+    # method to print the linkedlist
+    def printll(self):
```
Comment (CaedenPH) [inline data_structures/linked_list/quicksort_linkedlist.py:30]:
```suggestion
    def print_linked_list(self):
```

File: data_structures/linked_list/quicksort_linkedlist.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+    # method to insert nodes at the start of linkedlist
+    def insert(self, new_data: int):
+        """
+        Pushes the element at the start of the linkedlist and returns None
+        """
+        new_node = Node(new_data)
+        new_node.next = self.head
+        self.head = new_node
+        return
```
Comment (CaedenPH) [inline data_structures/linked_list/quicksort_linkedlist.py:27]:
```suggestion
```
Unecessary

File: data_structures/linked_list/quicksort_linkedlist.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        example: 1 ->3 ->4 ->2 ->9 ->6 ->8 -> 7
+        """
+        temp = self.head
+        if temp == None:
+            return "Linked List is empty"
+        while temp.next:
+            print(temp.data, "->", end="")
+            temp = temp.next
+        print(temp.data)
+        return
```
Comment (CaedenPH) [inline data_structures/linked_list/quicksort_linkedlist.py:42]:
```suggestion
```
Unecessary

File: data_structures/linked_list/quicksort_linkedlist.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+    # method to print the linkedlist
+    def printll(self):
+        """
+        Prints the linkedlist in the particular format and returns None
+        example: 1 ->3 ->4 ->2 ->9 ->6 ->8 -> 7
+        """
+        temp = self.head
+        if temp == None:
```
Comment (CaedenPH) [inline data_structures/linked_list/quicksort_linkedlist.py:36]:
```suggestion
        if temp is None:
```

File: data_structures/linked_list/quicksort_linkedlist.py [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+            return "Linked List is empty"
```
Comment (CaedenPH) [inline data_structures/linked_list/quicksort_linkedlist.py:37]:
```suggestion
            return
```
Returning `str` is a bad idea - Either raise an error, return, or print that it is empty

File: data_structures/linked_list/quicksort_linkedlist.py [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+        print(temp.data)
+        return
+
+
+def partition_algo(start: Node, end: Node):
+    """
+    This function takes in two pointers and returns the correct position of the pivot
+    in the linkedlist between the given two pointers
+    """
+    if start == None or start.next == None:
```
Comment (CaedenPH) [inline data_structures/linked_list/quicksort_linkedlist.py:50]:
```suggestion
    if start is None or start.next is None:
```

File: data_structures/linked_list/quicksort_linkedlist.py [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+# recursive quicksort for function calls
+def quicksort_ll(start: Node, end: None):
+    """
+    This function sorts the linkedlist using quicksort and returns None
+    """
+    if start != end:
+        pos = partition_algo(start, end)
+        quicksort_ll(start, pos)
+        quicksort_ll(pos.next, end)
+        return
```
Comment (CaedenPH) [inline data_structures/linked_list/quicksort_linkedlist.py:76]:
```suggestion
```
Again, unnecessary

**Review by CaedenPH — CHANGES_REQUESTED** [review]
Please typehint every function parameter and return.
Please also add [doctests](https://docs.python.org/3/library/doctest.html)

File: data_structures/linked_list/quicksort_linkedlist.py [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+
+    # method to insert nodes at the start of linkedlist
+    def insert(self, new_data: int):
+        """
+        Pushes the element at the start of the linkedlist and returns None
+        """
+        new_node = Node(new_data)
+        new_node.next = self.head
+        self.head = new_node
+        return
```
Comment (rohan472000) [inline data_structures/linked_list/quicksort_linkedlist.py:27]: For the first issue in `ruff` failure, you can simply remove the `return` statement as it is not doing anything useful.

File: data_structures/linked_list/quicksort_linkedlist.py [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+
+    # method to print the linkedlist
+    def printll(self):
+        """
+        Prints the linkedlist in the particular format and returns None
+        example: 1 ->3 ->4 ->2 ->9 ->6 ->8 -> 7
+        """
+        temp = self.head
+        if temp == None:
```
Comment (rohan472000) [inline data_structures/linked_list/quicksort_linkedlist.py:36]: you should replace the comparison to `None` with `is None` in line-36.

File: data_structures/linked_list/quicksort_linkedlist.py [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+    if start != end:
+        pos = partition_algo(start, end)
+        quicksort_ll(start, pos)
+        quicksort_ll(pos.next, end)
+
+
+if __name__ == "__main__":
+    ll = LinkedList()
+    print(
+        "Enter the space seperated values of numbers to be inserted in linkedlist prompted below:"
```
Comment (rohan472000) [inline data_structures/linked_list/quicksort_linkedlist.py:81]: For the line length issue in line-81, you can split the line into two or more shorter lines

**Review by rohan472000 — CHANGES_REQUESTED** [review]

**Comment by tianyizheng02** [comment]
Closing due to lack of response and lack of unit tests.

**Pull request closed without merging**